### DISCUSSION TITLE
machine.freq() range ?

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#11537] [Hardware & Peripherals]
I'm trying to knock a few mA off a 36mA lora equipped TTGO esp32 board. It's setup for 160 MHz default and the current drops to 30mA if I set it to 80MHz. When I push it down to 40MHz the current drops to 21mA which is nice but I lose the repl. I guess I could continue development at 80MHz & only push it down to 40 MHz when I'm finished but I'm worried I might be pushing my luck.

Anybody hazard a guess why the repl fails at quarter speed & what other wheels might also fall off at that rate?

**Comment by jimmo** [comment]
This is S3/C3-specific but might give you some hints: https://github.com/micropython/micropython/pull/11490

**Reply by kjm1102** [reply]
That looks like something that has to be done in C & then baked into a custom version of upython? 

Is there a way to find out what flavour of esp32 is on this particular board Jim? os.uname() simply says machine='ESP32 module with ESP32' but would be nice to discover what type of esp32 it is.

**Reply by bixb922** [reply]
With the ESP32-S3, os.uname().machine contains "ESP32S3".

**Reply by kjm1102** [reply]
tnx for that, gives
```>>> os.uname().machine
'ESP32 module with ESP32'
```
for this TTGO board so I'm assuming it must be an early model dual core chip.

**Reply by davefes** [reply]
`esptool.py  --chip esp32 chip_id (shows the chip version)`
